Sources/Header.swift

```diff
@@ -0,0 +1,20 @@
+/// The Header Object follows the structure of the Parameter Object with the following changes:
+/// - 'name' MUST NOT be specified, it is given in the corresponding headers map.
+/// - 'in' MUST NOT be specified, it is implicitly in header.
+/// All traits that are affected by the location MUST be applicable to a location of header (for example, style).
+
+public typealias Header = ParameterDefinition
+typealias HeaderBuilder = ParameterDefinitionBuilder
+
+extension HeaderBuilder {
+    static func resolve(_ swagger: SwaggerBuilder, reference: Reference<HeaderBuilder>) throws
+        -> Either<Header, Structure<Header>>
+    {
+        switch reference {
+        case .pointer(let pointer):
+            return .b(try self.resolver.resolve(swagger, pointer: pointer))
+        case .value(let builder):
+            return .a(try builder.build(swagger))
+        }
+    }
+}
```

Comment:
### [Header Object](https://swagger.io/specification/#headerObject)

This is quite the same as Parameter in OAS3, except that it excludes (with MUST) the 'name' and 'in' fields. Thus I've created the ParameterDefinition and it's builder which are reused here and in Parameter itself.